src/rgw/rgw_common.h

```diff
@@ -796,11 +796,15 @@ struct RGWUserInfo
       type(TYPE_NONE) {
   }
 
-  RGWAccessKey* get_key0() {
+  RGWAccessKey* get_key(const string& access_key) {
     if (access_keys.empty())
       return nullptr;
+
+    auto k = access_keys.find(access_key);
```

Comment:
Maybe you could check for an empty access key and return the first key, else follow this logic?